### PR TITLE
Code inspection warning fixes

### PR DESCRIPTION
**Pull request by abhipsaMisra**
The individual commit has the details of the warning fixed

### CODE REVIEW

File: device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsSenderLinkHandler.java [hunk 1]
```diff
@@ -72,7 +72,7 @@ public void onDelivery(Event event)
         //Safe to cast here because this callback will only ever fire for acknowledgements received on this sender link
         Delivery delivery = event.getDelivery();
 
-        int deliveryTag = Integer.valueOf(new String(event.getDelivery().getTag()));
+        int deliveryTag = Integer.parseInt(new String(event.getDelivery().getTag()));
```
Comment (abhipsaMisra) [inline device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsSenderLinkHandler.java:75]: Integer.valueOf() uses Integer.parseInt() internally

File: device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceClient.java [hunk 2]
```diff
@@ -640,14 +640,16 @@ public void getDeviceTwin() throws IOException
...
+     * @param <Type1> The type of the desired property key. Since the twin is a json object, the key will always be a String.
+     * @param <Type2> The type of the desired property value.
      *
      * @throws IllegalArgumentException if the callback is {@code null}
      * @throws UnsupportedOperationException if called more than once on the same device
      * @throws IOException if called when client is not opened
      */
-    public void startDeviceTwin(IotHubEventCallback deviceTwinStatusCallback, Object deviceTwinStatusCallbackContext,
-                                        PropertyCallBack genericPropertyCallBack, Object genericPropertyCallBackContext)
+    public <Type1, Type2> void startDeviceTwin(IotHubEventCallback deviceTwinStatusCallback, Object deviceTwinStatusCallbackContext,
```
Comment (timtay-microsoft) [inline device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceClient.java:651]: Is this a breaking change?

Comment (abhipsaMisra) [reply device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceClient.java:651]: I haven't updated any tests, so I think it shouldn't be a breaking change.

Comment (abhipsaMisra) [reply device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceClient.java:651]: If any of the tests break I'll revert this change.

Comment (timtay-microsoft) [reply device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceClient.java:651]: I spent a minute or two playing with the twin samples after using this change, and nothing seems to be broken, but the feature does seem to be enhanced which is nice

Comment (abhipsaMisra) [reply device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceClient.java:651]: cool, thanks for testing it out Tim!

**Review by timtay-microsoft — APPROVED** [review]

**Pull request merged**